chains/manager.go

```diff
@@ -418,7 +430,8 @@ func (m *manager) createAvalancheChain(
 	// VM uses this channel to notify engine that a block is ready to be made
 	msgChan := make(chan common.Message, defaultChannelSize)
 
-	if err := vm.Initialize(ctx, vmDB, genesisData, msgChan, fxs); err != nil {
+	chainConfig := m.ChainConfigs[ctx.ChainID]
+	if err := vm.Initialize(ctx, vmDBManager, genesisData, chainConfig.Upgrades, chainConfig.Settings, msgChan, fxs); err != nil {
```

Comment:
If we're passing in both fields on `chainConfig`, why not parameterize `Initialize` on `chainConfig` instead of both its fields?

Reply:
I think the interface for VM initialize is cleaner this way. ChainConfig is currently defined in a place where these VMs are also imported, which would create a circular dependency. It can be moved and re-organized, but I think that using byte arrays as the param type here keeps it much simpler.